[9.x.x] isSceneCamera changed to property, isPreviewCamera and cameraType added plus two bugfixes for 1240723 & 1204376

ellioman opened the pull request:
# Purpose of this PR
This PR does the following:
- Makes the isSceneCamera a property in renderingData.cameraData struct
- Adds cameraType
- Adds isPreviewCamera
- Fixes two issues reported by users

## Issue 1240723
[Grid lines were being drawn on top of opaque objects in the preview window when looking at a model when depth or opaque toggles were enabled in the URP Asset.](https://issuetracker.unity3d.com/issues/urp-grid-is-rendered-in-front-of-the-model-in-the-inspector-animation-preview-window-when-depth-or-opaque-texture-is-enabled) 

**The fix is to** force a depth prepass and disable opaque textures when rendering with a preview camera.

**Before:**
<img width="345" alt="Before" src="https://user-images.githubusercontent.com/225349/80220801-adccce00-8644-11ea-9f3c-6a5abc24d846.png">

**After:**
<img width="345" alt="After" src="https://user-images.githubusercontent.com/225349/80220829-b6bd9f80-8644-11ea-8b16-b389abb4f74d.png">

## Issue 1204376
[The layermask in the renderer was affecting what was being rendered in the preview camera](https://issuetracker.unity3d.com/issues/urp-prefab-preview-is-blank-when-a-custom-forward-renderer-data-and-default-layer-mask-is-mixed-are-used). If the user disabled some layers then the object would disappear in the preview window. 

**The fix is to** force the layer mask to Everything when drawing objects in the preview window.


---

### Checklist for PR maker
- [x] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---

### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)

**Yamato**:
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fpreview-window-gridlines-issue

phi-lira commented on com.unity.render-pipelines.universal/Runtime/UniversalRenderPipeline.cs:
```diff
@@ -433,6 +433,7 @@ static void InitializeStackedCameraData(Camera baseCamera, UniversalAdditionalCa
             cameraData.targetTexture = baseCamera.targetTexture;
             cameraData.isStereoEnabled = IsStereoEnabled(baseCamera);
             cameraData.isSceneViewCamera = baseCamera.cameraType == CameraType.SceneView;
+            cameraData.isPreviewCamera = baseCamera.cameraType == CameraType.Preview;
```
hm, we should probably add to cameraData the camera type isntead. otherwise we might be in a situation that we add every single type.

ellioman replied:
@phi-lira Yeah, that's a good point. 
The question then is: Should we then remove or keep the isSceneViewCamera variable?

ellioman commented:
@phi-lira I've now marked isSceneCamera as obsolete, added cameraType and fixed the code where isSceneCamera was being used.

I've also now added another bugfix in this PR as it also deals with the preview camera.
Do me a favour and take another look for me.

phi-lira commented on com.unity.render-pipelines.universal/Runtime/UniversalRenderPipelineCore.cs:
```diff
@@ -95,7 +95,8 @@ public Matrix4x4 GetGPUProjectionMatrix()
         public float renderScale;
         public bool clearDepth;
-        public bool isSceneViewCamera;
+        [Obsolete("Please use cameraType instead.", true)] public bool isSceneViewCamera;
```
This just occurred to me, what if we make this a property instead that returns `cameraType == SceneView`? We don't consume data from struct and avoid duplicating the code in several places.
